app/scripts/views/sign_up.js

```diff
@@ -51,6 +51,17 @@ function (Cocktail, _, p, BaseView, FormView, Template, Session, AuthErrors,
         return p(false);
       }
 
+      /**
+       * check if there is an account we can suggest for an OAuth flow.
+       */
+      var account = this.user.getChooserAccount();
+      var serviceName = this.relier.get('serviceName');
+
+      if (account.get('email') && serviceName) {
```

Comment:
We'll need a flag here also to disable the redirection when the user clicks "Create an account" from /signin. Something like `! relier.has("force_signup")`.